src/Pact/Analyze/Eval/Term.hs

```diff
@@ -288,6 +290,32 @@ readFields tn sRk tid (SObjectUnsafe (SCons' sym fieldType subSchema)) = do
     AVal _ _ -> error
       "impossible: unexpected type of cell provenance in readFields"
 
+mkChainData :: SingTy ('TyObject ty) -> Analyze (S (ConcreteObj ty))
+mkChainData (SObjectUnsafe SNil') = pure $ sansProv $ literal ()
```

Comment:
I recommend using `foldrSObject` so you don't have to use `SObjectUnsafe` (unless there's a reason I didn't notice that you can't)

Reply:
Unfortunately it seems like we need to do raw recursion here because we need access to the pre-transformed `subSchema` in order to be able to construct `subObjTy`